### ISSUE TITLE
BeastTreeFromMaster using taxonset instead of alignment input

### ISSUE THREAD

**Issue by rbouckaert** [tgvaughan/MASTER#50]
When running the `examples/BeastTreeFromMasterExample.xml` things are fine, but adding a taxonset

```
    <taxonset id="taxa" spec="beast.evolution.alignment.TaxonSet">
    	<taxon spec="beast.evolution.alignment.Taxon" id="human"/>
    	<taxon spec="beast.evolution.alignment.Taxon" id="chimp"/>
    	<taxon spec="beast.evolution.alignment.Taxon" id="bonobo"/>
    	<taxon spec="beast.evolution.alignment.Taxon" id="gorilla"/>
    	<taxon spec="beast.evolution.alignment.Taxon" id="orangutan"/>
    	<taxon spec="beast.evolution.alignment.Taxon" id="siamang"/>
    </taxonset>
```

and replacing

`<input spec='BeastTreeFromMaster' alignment="@alignment" reverseTime="true" id='tree'>`

with

`<input spec='BeastTreeFromMaster' taxonset="@taxa" reverseTime="true" id='tree'>`

gives a incorrectly calculated likelihood error at sample 10000 with seed 1594066944914.

```
P(posterior) = -1827.6007291850156 (was -1716.4030498385023)  **
	P(coalescent) = 7.797781122413868 (was 7.797781122413868)
	P(treeLikelihood) = -1835.3985103074294 (was -1724.2008309609162)  **
At sample 10000
Likelihood incorrectly calculated: -1716.4030498385023 != -1827.6007291850156(111.19767934651327) Operator: Uniform()
```

Adding the lines 

```
            } else if (m_taxonset.get() != null) {
            	leaf.setNr(m_taxonset.get().getTaxonIndex(leaf.getID()));
```

to the `initNodeNumbers()` method to process the taxon set if specified fixes this. However,  I wonder whether this is a sign of something more sinister, since I cannot quite understand why changing node numbers in the tree fixes the problem.

**Comment by tgvaughan** [comment]
Thanks Remco!  I've patched this in f32fb4d and will issue a release.  I'm also confused about how the particular error you reported manifests.  It must be that some other class is paying attention to the taxon indices indicated by the TaxonSet while others are looking directly at the result of node.getNr()?  In any case, these should always be consistent, so the fault definitely lies with this class.

**Comment by rbouckaert** [comment]
Thanks for the quick fix! :-)
